### PR TITLE
UI: Update template linting and fix missed curly invocations

### PR DESCRIPTION
**Pull request by backspace**
On a first pass at running the `no-implicit-this` codemod I noticed a few remaining curly bracket invocations. I looked for a template linting rule that would catch these and found that it wasn’t added to `ember-template-lint` [until 1.5.0](https://github.com/ember-template-lint/ember-template-lint/blob/master/CHANGELOG.md#v150-2019-07-31), we had 1.2.0. `ember-cli-template-lint` is [now deprecated](https://github.com/ember-template-lint/ember-cli-template-lint/commit/71e3080858c84eccdb9e2bb7cec4d162c39bc1b4#diff-04c6e90faac2675aa89e2176d2eec7d8), and while the Ember CLI blueprint wasn’t [updated until 3.17](https://github.com/ember-cli/ember-cli/commit/1dd8f4b8c08488d74dbc25fd2898339e3b0419c4), I think it makes sense to migrate away regardless.

The `no-curly-component-invocation` rule sadly missed a few invocations that I fixed manually, but I think it still makes sense to turn on the rule to catch future additions when possible.

Updating the template lint dependency included updates to the `recommended` set of rules, so I addressed those in individual commits. The only one I have any concern about is the `tabindex=1` changing to `tabindex=0` in `MultiSelectDropdown` due to the [`no-positive-tabindex` rule](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-positive-tabindex.md), but I tested tabbing with the keyboard after the change and it acted as I would expect 🤞

(This is set to merge into #8094 for now as it’s branched off it.)

### CODE REVIEW

**Comment by backspace** [comment]
hmm… I’m guessing that test audit difference means that template linting isn’t being included in the tests for some reason, I’ll check that out 🤔

**Comment by backspace** [comment]
hmm yes it appears that [linting is no longer part of `ember test`](https://github.com/ember-cli/ember-cli/pull/9009#issuecomment-590015616). It’s subpar but I think it’s okay to have this one-time drop in the apparent number of tests.

**Comment by DingoEatingFuzz** [comment]
> The only one I have any concern about is the tabindex=1 changing to tabindex=0 in MultiSelectDropdown due to the no-positive-tabindex rule, but I tested tabbing with the keyboard after the change and it acted as I would expect 🤞

Yeah, I'm sure that's fine. The only reason for the manual `tabindex` values in the first place was to make the `li`s tabbable and the `input`s non-tabbable for `:focus` style reasons. There is no order trickery going on, so setting everything to 0 here should be the functional equivalent as setting everything to 1.

File: ui/tests/integration/list-pagination-test.js [hunk 1]
```diff
@@ -20,20 +20,20 @@ module('Integration | Component | list pagination', function(hooks) {
   test('the source property', async function(assert) {
     this.set('source', list100);
     await render(hbs`
-      {{#list-pagination source=source as |p|}}
+      <ListPagination @source={{source}} as |p|>
         <span class="page-info">{{p.currentPage}} of {{p.totalPages}}</span>
-        {{#p.first}}<span class="first">first</span>{{/p.first}}
-        {{#p.prev}}<span class="prev">prev</span>{{/p.prev}}
+        <p.first><span class="first">first</span></p.first>
+        <p.prev><span class="prev">prev</span></p.prev>
```
Comment (DingoEatingFuzz) [inline ui/tests/integration/list-pagination-test.js:26]: I think at some point we should refactor contextual components to yield components using the same capitalized camelcase as is now standard for component invocation.

**Comment by backspace** [comment]
> Yeah, I'm sure that's fine. The only reason for the manual `tabindex` values in the first place was to make the `li`s tabbable and the `input`s non-tabbable for `:focus` style reasons. There is no order trickery going on, so setting everything to 0 here should be the functional equivalent as setting everything to 1.

Excellent, thanks for the confirmation 😀

File: ui/app/templates/clients/client/index.hbs [hunk 2]
```diff
@@ -74,7 +74,7 @@
           <p data-test-message>Allocations have been drained and the node has been set to ineligible.</p>
         </div>
         <div class="column is-centered is-minimum">
-          <button data-test-dimiss class="button is-info" onclick={{action (mut showDrainNotification) false}}>Okay</button>
+          <button data-test-dimiss class="button is-info" onclick={{action (mut showDrainNotification) false}} type="button">Okay</button>
```
[the remark this answers is not in the document]
Comment (backspace) [reply ui/app/templates/clients/client/index.hbs:77]: I actually didn’t even know this, or forgot, so I’m glad too!

File: ui/tests/integration/list-pagination-test.js (hunk 1 above)
Comment (backspace) [reply ui/tests/integration/list-pagination-test.js:26]: Agreed!

**Pull request merged**